test/libraries/SafeMigration.spec.ts

```diff
@@ -0,0 +1,352 @@
+                it("reverts if not called via delegatecall", async () => {
+                    const {
+                        safe,
+                        migration,
+                        signers: [user1],
+                    } = await setupTests();
+                    await expect(
+                        executeContractCallWithSigners(safe, migration, "migrateSingleton", [], [user1], false),
+                    ).to.be.revertedWith("GS013");
+                });
+
+                it("migrates the singleton", async () => {
+                    const {
+                        safe,
+                        migration,
+                        signers: [user1],
+                    } = await setupTests();
+                    const safeAddress = await safe.getAddress();
+                    // The emit matcher checks the address, which is the Safe as delegatecall is used
+                    const migrationSafe = migration.attach(safeAddress);
+
+                    await expect(executeContractCallWithSigners(safe, migration, "migrateSingleton", [], [user1], true))
+                        .to.emit(migrationSafe, "ChangedMasterCopy")
+                        .withArgs(SAFE_SINGLETON_ADDRESS);
+
+                    const singletonResp = await user1.call({ to: safeAddress, data: migratedInterface.encodeFunctionData("masterCopy") });
+                    expect(migratedInterface.decodeFunctionResult("masterCopy", singletonResp)[0]).to.eq(SAFE_SINGLETON_ADDRESS);
+                });
+
+                it("doesn't touch important storage slots", async () => {
+                    const {
+                        safe,
+                        migration,
+                        signers: [user1],
+                    } = await setupTests();
+                    const safeAddress = await safe.getAddress();
+
+                    const ownerCountBeforeMigration = await hre.ethers.provider.getStorage(safeAddress, 3);
+                    const thresholdBeforeMigration = await hre.ethers.provider.getStorage(safeAddress, 4);
+                    const nonceBeforeMigration = await hre.ethers.provider.getStorage(safeAddress, 5);
+                    const guardBeforeMigration = await hre.ethers.provider.getStorage(safeAddress, GUARD_STORAGE_SLOT);
+                    const fallbackHandlerBeforeMigration = await hre.ethers.provider.getStorage(safeAddress, FALLBACK_HANDLER_STORAGE_SLOT);
+
+                    expect(await executeContractCallWithSigners(safe, migration, "migrateSingleton", [], [user1], true));
+
+                    expect(await hre.ethers.provider.getStorage(safeAddress, 3)).to.be.eq(ownerCountBeforeMigration);
+                    expect(await hre.ethers.provider.getStorage(safeAddress, 4)).to.be.eq(thresholdBeforeMigration);
+                    expect(await hre.ethers.provider.getStorage(safeAddress, 5)).to.be.eq(
+                        ethers.toBigInt(nonceBeforeMigration) + ethers.toBigInt(1),
```

Comment:
ubernit: I _think_ you can just write this as:

```suggestion
                        BigInt(nonceBeforeMigration) + 1n,
```

(note that `n` suffix to numerical literals is for creating `bigint`s in JS).

Reply:
This also appears in other places.